### ISSUE TITLE
applyDefaultParams() not found in RouteOptions.Builder()

### ISSUE THREAD

**Issue by Gango98** [mapbox/mapbox-navigation-android#4932]
I'm following the official Mapbox Documentation to build and Android app.

I'm adding the navigation section following this example https://docs.mapbox.com/android/navigation/examples/basic-nav-sdk-only/

In this block of code the method applyDefaultParams() is not found

```
mapboxNavigation.requestRoutes(
    RouteOptions.builder()
    .applyDefaultParams() // <- NOT FOUND
    .accessToken(ACCESS_TOKEN)
    .coordinates(pointList)
    .alternatives(true)
    .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
    .build(),
  routesReqCallback
);
```

When I run the code without this line I get the following error: Missing required properties: baseUrl user requestUuid

Any suggestions? Am I doing something wrong? Are the docs wrong? Are there any other methods to request a route in Mapbox?

**Comment by Guardiola31337** [comment]
Hey @Gango98 thanks for reaching out and report your question.

You're right, that changed in https://github.com/mapbox/mapbox-navigation-android/pull/4320 We included a warning in the Release notes https://github.com/mapbox/mapbox-navigation-android/releases/tag/v2.0.0-beta.10

> * :warning: Requesting a route via `MapboxNavigation#requestRoutes` does not automatically append any defaults. Defaults are now available under these extensions:
>    - `RouteOptions.Builder.applyDefaultNavigationOptions()` that applies the options that are required for the route request to execute or otherwise recommended for the Navigation SDK and all of its features to provide the best car navigation experience.
>    - `RouteOptions.Builder.applyLanguageAndVoiceUnitOptions(context: Context)` that applies the options that adapt the returned instructions' language and voice unit based on the device's `Locale`.
The extensions are very much recommended to be called when building the `RouteOptions` object and then tweaked for a specific use case. ([#4320](https://github.com/mapbox/mapbox-navigation-android/pull/4320))

but it's true that docs are not updated yet (we're working on it). For now, `applyDefaultNavigationOptions` should do the trick. Hope that helps a bit!

Thanks again!

**Comment by Gango98** [comment]
Thank you for your fast reply.

I found this solution in other examples online but also .applyDefaultNavigationOptions() is not found.
I'm using Mapbox Navigation 1.5.1 but I also tried with v2.0.0

Here is my Mapbox definition in app-level gradle. Maybe it can be useful.

```
    implementation 'com.mapbox.mapboxsdk:mapbox-android-sdk:9.6.2'
    implementation ("com.mapbox.navigation:ui:1.5.1", {
        exclude group: 'com.mapbox.mapboxsdk', module:'mapbox-android-telemetry'
    })
```

**Comment by Guardiola31337** [comment]
Yeah in `v1.5.1` we had `applyDefaultParams` https://github.com/mapbox/mapbox-navigation-android/blob/v1.5.1/libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/extensions/RouteOptionsUtils.kt (if you're using Java it'd be something around these lines `MapboxRouteOptionsUtils.applyDefaultParams(RouteOptions.builder()).build();`) but that's internal (you shouldn't rely on anything `internal`). For reference, here's a Java example that doesn't use `applyDefaultParams` https://github.com/mapbox/mapbox-navigation-android/blob/v1.5.1/examples/src/main/java/com/mapbox/navigation/examples/core/NavigationMapRouteActivity.java and you can see what are the required options.

Could you share you `2.0` route request setup? You shouldn't have any problems with `applyDefaultNavigationOptions` (if you're using Java you should get it as `RouteOptionsExtensions.applyDefaultNavigationOptions(builder);`). At this point we encourage you to upgrade to `2.0` if possible, as GA is around the corner 🚀

**Comment by Gango98** [comment]
Ok, in `v1.5.1` the Java example that doesn't use `applyDefaultParams` works.

I tried to use `2.0` route request but I had a similar problem.
After setting up gradle files I tried to follow this page https://docs.mapbox.com/android/beta/navigation/guides/initialization/ but the first step doesn't work. Methods are not found

```
val navigationOptions = NavigationOptions.Builder(this)
    .accessToken("YOUR_ACCESS_TOKEN") // NOT FOUND
    .build() // NOT FOUND
```

**Comment by Gango98** [comment]
I'm using `2.0.0-rc.4`

Thank you. Based on your advice I found [this example ](https://github.com/mapbox/mapbox-navigation-android/blob/9ea951ef64f74fd76ff3d1ae36445f01e554e1c7/qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/MapboxRouteLineActivity.java) that helped me init `MapboxNavigation` and requesting routes.

The issue can be closed but I think that the documentation is not complete (especially for Java). I had to ask here and looking for random lines of code around many examples in this repo to make things working. Anyway, thank you for your help

**Comment by Guardiola31337** [comment]
Glad to hear you got it working 🎉 

> I think that the documentation is not complete (especially for Java).

Thanks for the feedback cc @colleenmcginnis @mapbox/docs 

Going to close here but don't hesitate to create a new issue if you have any other questions. Thanks again!
